integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/InsertIntoDirVisitor.java

```diff
@@ -27,12 +32,21 @@ public InsertIntoDirVisitor(OpenLineageContext context) {
   public List<OpenLineage.OutputDataset> apply(LogicalPlan x) {
     InsertIntoDir cmd = (InsertIntoDir) x;
     Optional<URI> optionalUri = ScalaConversionUtils.asJavaOptional(cmd.storage().locationUri());
+
     return optionalUri
         .map(
             uri ->
-                Collections.singletonList(
-                    outputDataset()
-                        .getDataset(PathUtils.fromURI(uri, "file"), cmd.child().schema())))
+                cmd.overwrite()
```

Comment:
Consider passing empty map to `getDataset` when `overwrite` is false to avoid repeated code.